client/src/header.rs

```diff
@@ -0,0 +1,100 @@
+use std::sync::Arc;
+
+use celestia_rpc::HeaderClient;
+use celestia_types::hash::Hash;
+use celestia_types::{ExtendedHeader, SyncState};
+use futures_util::{Stream, StreamExt};
+
+use crate::Result;
+use crate::client::Context;
+
+/// Header API for quering bridge nodes.
+pub struct HeaderApi {
+    ctx: Arc<Context>,
+}
+
+impl HeaderApi {
+    pub(crate) fn new(ctx: Arc<Context>) -> HeaderApi {
+        HeaderApi { ctx }
+    }
+
+    /// Returns the head from the node's header store.
+    // TODO: keep this name or `local_head`?
```

Comment:
It's a touch unintitive, since it _is_ `local_head`, but the _local_ is over there, on the node we're connecting. I think I'd keep it as `local_head` or `node_local_head`, but that's mostly cause I dislike bare `head` as too vague.

Reply:
in my case, i don't like `local_head` or `node_local_head`. even `node_head` makes it strange